File: src/main/java/com/truveta/opentoken/attributes/validation/NotNullOrEmptyValidator.java
/**
 * Copyright (c) Truveta. All rights reserved.
 */
package com.truveta.opentoken.attributes.validation;

/**
 * A Validator that asserts the value is <b>NOT</b> <code>null</code>
 * and blank.
 */
public final class NotNullOrEmptyValidator implements AttributeValidator {

    /**
     * Validates that the attribute value is not <code>null</code> or blank.
     */
    @Override
    public boolean eval(String value) {
        return value != null && !value.isBlank();
    }

}

File: lib/nodejs/opentoken/src/attributes/validation/NotNullOrEmptyValidator.ts
/**
 * Copyright (c) Truveta. All rights reserved.
 */

import { SerializableAttributeValidator } from './SerializableAttributeValidator';

/**
 * A Validator that asserts the value is **NOT** `null` and blank.
 */
export class NotNullOrEmptyValidator implements SerializableAttributeValidator {
  /**
   * Validates that the attribute value is not `null` or blank.
   *
   * @param value - The value to validate.
   * @returns true if the value is not null and not blank; false otherwise.
   */
  validate(value: string | null | undefined): boolean {
    return value != null && value.trim() !== '';
  }
}
